src/Diagrams/Path.hs

```diff
@@ -90,12 +94,16 @@ import           Linear.Vector
 --   and they form a monoid under /superposition/ (placing one path on
 --   top of another) rather than concatenation.
 newtype Path v n = Path [Located (Trail v n)]
-  deriving (Semigroup, Monoid
+  deriving (Semigroup, Monoid, Generic
```

Comment:
Do we need to check the GHC version for these `deriving Generic` statements? I read at https://ocharles.org.uk/blog/posts/2014-12-16-derive-generic.html that DeriveGeneric has only been in GHC since 7.2.